Server/src/utils/focus_nudge.py

```
@@ -255,8 +289,22 @@ def _focus_app_macos(app_name: str, unity_project_path: str | None = None) -> bo
                 # No project path provided - activate any Unity process
                 return _focus_any_unity_macos()
         else:
-            # For other apps, use direct activation
-            # Escape double quotes in app_name to prevent AppleScript injection
+            # For non-Unity apps, prefer bundle_id to avoid the Electron bug:
+            # VS Code's process name is "Electron", and `tell application "Electron"`
+            # can launch a standalone Electron instance instead of returning to VS Code.
+            if bundle_id:
+                escaped_bundle_id = bundle_id.replace('"', '\\"')
+                result = subprocess.run(
```

Ruff (0.15.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)